editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -156,9 +160,20 @@ private void installDefaultFacets(IProject project, MavenProject mavenProject, I
 		String camelVersion = getCamelVersion(mavenProject);
 		if (camelVersion != null) {
 			IFacetedProjectWorkingCopy fpwc = fproj.createWorkingCopy();
-			installFacet(fproj, fpwc, javaFacet, javaFacet.getLatestVersion(), monitor);
-			installFacet(fproj, fpwc, m2eFacet, m2eFacet.getLatestVersion(), monitor);
-			installFacet(fproj, fpwc, utilFacet, utilFacet.getLatestVersion(), monitor);
+			
+			// adjust facets we install based on the packaging type we find
+			String packaging = null;
+			if (mavenProject.getPackaging() != null) {
+			    packaging = mavenProject.getPackaging().toUpperCase();
+			}
+            installFacet(fproj, fpwc, javaFacet, javaFacet.getLatestVersion(), monitor);
+            installFacet(fproj, fpwc, m2eFacet, m2eFacet.getLatestVersion(), monitor);
+			if (packaging != null && WAR_PACKAGE.contentEquals(packaging)) {
```

Comment:
the != null check seems to be overhead here. btw why do u use contentEquals instead of equals(IgnoreCase) ?